cmd/skopeo/login.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"io"
+	"os"
+
+	"github.com/containers/common/pkg/auth"
+	"github.com/containers/image/v5/types"
+	"github.com/spf13/cobra"
+)
+
+type loginOptions struct {
+	global    *globalOptions
+	loginOpts auth.LoginOptions
+	getLogin  optionalBool
+	tlsVerify optionalBool
+}
+
+func loginCmd(global *globalOptions) *cobra.Command {
+	opts := loginOptions{
+		global: global,
+		loginOpts: auth.LoginOptions{
+			Stdin:  os.Stdin,
+			Stdout: os.Stdout,
+		},
+	}
+	cmd := &cobra.Command{
+		Use:     "login",
+		Short:   "Login to a container registry",
+		Long:    "Login to a container registry on a specified server.",
+		RunE:    commandAction(opts.run),
+		Example: `skopeo login quay.io`,
+	}
+	adjustUsage(cmd)
+	flags := cmd.Flags()
+	optionalBoolFlag(flags, &opts.getLogin, "get-login", "Return the current login user for the registry")
+	optionalBoolFlag(flags, &opts.tlsVerify, "tls-verify", "require HTTPS and verify certificates when accessing the registry")
+	flags.AddFlagSet(auth.GetLoginFlags(&opts.loginOpts))
+	return cmd
+}
+
+func (opts *loginOptions) run(args []string, stdout io.Writer) error {
+	ctx, cancel := opts.global.commandTimeoutContext()
+	defer cancel()
+	sys := opts.global.newSystemContext()
+	sys.DockerCertPath = opts.loginOpts.CertDir
```

Comment:
Nothing in c/common documents that the caller is responsible for handling this. Sigh, the `LoginOpts` members are used in way too many different ways.

Reply:
Consider https://github.com/containers/common/pull/141 .